File: src/main/java/rr/language/completion/providers/RRCompletionProvider.java
package rr.language.completion.providers;

import com.intellij.codeInsight.completion.CompletionParameters;
import com.intellij.codeInsight.completion.CompletionProvider;
import com.intellij.codeInsight.completion.CompletionResultSet;
import com.intellij.util.ProcessingContext;
import org.jetbrains.annotations.NotNull;


/* Extended so that visibility of addCompletions can be made public, so that Providers can call other Providers.*/
public class RRCompletionProvider extends CompletionProvider<CompletionParameters> {
    @Override
    public void addCompletions(@NotNull CompletionParameters parameters, @NotNull ProcessingContext context, @NotNull CompletionResultSet resultSet) {
    }
}

File: src/main/java/rr/language/completion/providers/RRCompletionProvider.kt
package rr.language.completion.providers

import com.intellij.codeInsight.completion.CompletionParameters
import com.intellij.codeInsight.completion.CompletionProvider
import com.intellij.codeInsight.completion.CompletionResultSet
import com.intellij.util.ProcessingContext

/* Extended so that visibility of addCompletions can be made public, so that Providers can call other Providers.*/
open class RRCompletionProvider : CompletionProvider<CompletionParameters>() {
    public override fun addCompletions(
        parameters: CompletionParameters,
        context: ProcessingContext,
        resultSet: CompletionResultSet
    ) {
    }
}